pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -195,6 +195,13 @@ class Timestamp(_Timestamp):
     nanosecond : int, optional, default 0
         .. versionadded:: 0.23.0
     tzinfo : datetime.tzinfo, optional, default None
+    fold : {0, 1}, default None
```

Comment:
Echoing some of @mroeschke's concerns, I think it doesn't actually make sense to specify `fold` if `ts_input` could have a valid value for `fold`. I think a useful rule would be to throw an error if `ts_input` corresponds to anything equivalent to an aware datetime, so `int`, `float` and aware `datetime`-like should all raise if `fold` is set, as should `str` if it parses to an aware datetime (e.g. `2019-01-01T00:00+04:00`).

I am not really sure what the best thing to do is if `datetime.datetime` is not aware and `fold=1` but `Timestamp`'s `fold=0`. That seems like a weird edge case and I suspect most people won't care if the response they get is, "don't do that", but maybe I'm wrong there.

Reply:
@pganssle 
Done. Now we raise on anything that could be unambiguous.